### ISSUE TITLE
`<label>` should not describe non-labelable `.form-control-static` elements

### ISSUE THREAD

**Issue by zacechola** [twbs/bootstrap#16305]
In the [static form control documentation](http://getbootstrap.com/css/#forms-controls-static), `label` describes nothing without a `for` present. Adding one and a corresponding ID to `.form-control-static` doesn't make sense, though, since it's a `p` element and not a form control. `p` [is not listed as a labelable element](http://www.w3.org/TR/html5/forms.html#category-label)

```html
<form class="form-horizontal">
  <div class="form-group">
    <label class="col-sm-2 control-label">Email</label>
    <div class="col-sm-10">
      <p class="form-control-static">email@example.com</p>
    </div>
  </div>
  <div class="form-group">
    <label for="inputPassword" class="col-sm-2 control-label">Password</label>
    <div class="col-sm-10">
      <input type="password" class="form-control" id="inputPassword" placeholder="Password">
    </div>
  </div>
</form>
```

This seems to be in some conflict with [H44](http://www.w3.org/TR/WCAG20-TECHS/H44) at least according to warnings thrown by some automated accessibility testing software, but I'm unsure about the practical implications here. Is this a job for `aria-labledby`?

```html
<div class="form-group">
  <span class="col-sm-2 control-label" id="email">Email</span>
  <div class="col-sm-10">
    <p class="form-control-static" aria-labeledby="email">email@example.com</p>
  </div>
</div>
```

cc @patrickhlauke

**Comment by patrickhlauke** [comment]
To be honest, the most appropriate way to deal with the specific example would be to use a read-only input for the static text
```
<div class="form-group">
  <label class="col-sm-2 control-label" for="inputUsername">Email</label>
  <div class="col-sm-10">
    <input type="text" readonly="" id="inputUsername" value="email@example.com">
  </div>
</div>
```
`aria-labelledby` is an interesting idea, but from testing something similar a while ago it seems no assistive tech actually does anything useful with this association for things that aren't normally labeled.

using `<label>` without an actual association is not really harmful, just mostly redundant...and as we don't (afaik) currently have a class that replicates the same styling as actual labels, i'd give that a pass.

**Comment by patrickhlauke** [comment]
before i dive into this, my plan would be to make a new style for inputs marked with `readonly` to mimic what we currently have associated with  `.form-control-static`, change the wording for "Static control" to actually say that read-only inputs should be used (rather than just text), and to change the code sample accordingly. /cc @mdo ?

**Comment by zacechola** [comment]
Thanks for the response. I had a feeling the aria method was a pipe dream. 

Anyway, is it maybe this simple? http://jsbin.com/fuluri/1

**Comment by patrickhlauke** [comment]
> Anyway, is it maybe this simple? http://jsbin.com/fuluri/1

in essence, yes, but that classname stuff is a bit fugly...should just be default for inputs with `readonly` etc, and the docs prose will need changing

**Comment by mdo** [comment]
I dig that for a v4 thing, @patrickhlauke. I knew when we made that static control we'd probably have some problems with the proper markup with regards to `<label>`s.

**Comment by patrickhlauke** [comment]
So...punting for now you mean? I can probably tackle a quick fix for next 3.x release...

**Comment by mdo** [comment]
I'd be interested in any kind of quick fixes for v3, but anything more—like changing from the `p.form-control-static` we use today to `input[readonly].form-control-static`—would be for v4 I'd think.

**Comment by patrickhlauke** [comment]
Ok, let's say we're punting on this for now then but will look at this properly in v4. In the meantime, I think the code as it currently stands in the documentation is not ideal, but if we don't want to move away from `p.form-control-static` just now, also not necessarily harmful.

**Comment by patrickhlauke** [comment]
Oh and I should say thanks @zacechola for raising this, and rest assured it'll be properly taken care of in v4, for sure.
